test/organization_test.go

```diff
@@ -122,7 +132,20 @@ modfor:
 	}
 
 	t.Log("update")
-	org.Labels = map[string]string{"aa": "bb"}
+	updateCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	t.Cleanup(cancel)
+	require.NoError(t, wait.PollUntil(time.Second, func() (done bool, err error) {
```

Comment:
please extract this to a separate function. It breaks the test reading when you have 10-15 code block lines for updating and handling possible modification errors.

Reply:
I don't think it is needed to split this into a function, this is just:
get organization,
update it with new label
Why do you think this breaks the test reading?

Reply:
The flakiness of e2e test is not introduced by this PR, the master branch e2e tests are already flaky, so moved this into dedicated PR #30 for it.